source/Playlist.cpp

```diff
@@ -0,0 +1,134 @@
+/* Playlist.cpp
+ Copyright (c) 2022 by RisingLeaf
+
+ Endless Sky is free software: you can redistribute it and/or modify it under the
+ terms of the GNU General Public License as published by the Free Software
+ Foundation, either version 3 of the License, or (at your option) any later version.
+
+ Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+ WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License along with
+ this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "DataNode.h"
+#include "GameData.h"
+#include "Playlist.h"
+#include "PlayerInfo.h"
+#include "System.h"
+
+#include <algorithm>
+
+using namespace std;
+
+
+
+Playlist::Playlist(const DataNode &node)
+{
+	Load(node);
+}
+
+
+
+void Playlist::Load(const DataNode &node)
+{
+	// All tracks need a name.
+	if(node.Size() < 2)
+	{
+		node.PrintTrace("Error: No name specified for playlist:");
+		return;
+	}
+
+	if(!name.empty())
```

Comment:
so you dont support modification of playlists by event, I guess we can always add it in another PR

Reply:
Yes, I think so

Reply:
I dont want to overload this one

Reply:
Yeah thats fair me neither